control/metrics.go

```diff
@@ -706,3 +563,123 @@ func addStandardAndWorkflowTags(m core.Metric, allTags map[string]map[string]str
 	}
 	return metric
 }
+
+// isTuple returns true when incoming namespace's element has been recognized as a tuple, otherwise returns false
+// notice, that the tuple is a string which starts with `core.TuplePrefix`, ends with `core.TupleSuffix` and contains at least one `core.TupleSeparator`
+// e.g. (host0;host1)
+func isTuple(element string) bool {
+	if strings.HasPrefix(element, core.TuplePrefix) && strings.HasSuffix(element, core.TupleSuffix) && strings.Contains(element, core.TupleSeparator) {
+		return true
+	}
+	return false
+}
+
+// containsTuple checks if a given element of namespace has a tuple; if yes, returns true and recognized tuple's items
+func containsTuple(nsElement string) (bool, []string) {
+	tupleItems := []string{}
+	if isTuple(nsElement) {
+		if strings.ContainsAny(nsElement, "*") {
+			// an asterisk covers all tuples cases (eg. /intel/mock/(host0;host1;*)/baz)
+			// so to avoid retrieving the same metric more than once, return only '*' as a tuple's items
+			tupleItems = []string{"*"}
+		} else {
+			tuple := strings.TrimSuffix(strings.TrimPrefix(nsElement, core.TuplePrefix), core.TupleSuffix)
+			tuple = strings.Replace(tuple, core.TupleSeparator, " ", -1)
```

Comment:
I added also small tests which cover that (see [control/metrics_small_test.go](https://github.com/intelsdi-x/snap/pull/1220/files#diff-d03b6fc79a4cf76dee147f92b0cd76ef))